File: week_1/day-5-string-hashing/first_non_repeating_char_follow_up_questions/firstUniqueCharacter.js
// Program: firstUniqueCharacter.js
function firstUniqueChar(s) {
    const freq = {};
    for (let ch of s) freq[ch] = (freq[ch] || 0) + 1;

    for (let ch of s) {
        if (freq[ch] === 1) return ch;
    }
    return null;
}

console.log(firstUniqueChar("leetcode")); // l

File: week_1/day-5-string-hashing/first_non_repeating_char_follow_up_questions/first_unique_character.py
# Program: first_unique_character.py
def firstUniqueChar(s):
    freq = {}
    for ch in s:
        freq[ch] = freq.get(ch, 0) + 1

    for ch in s:
        if freq[ch] == 1:
            return ch
    return None

print(firstUniqueChar("leetcode"))  # Output: l
